cpu/atmega_common/periph/gpio_ll.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2015 HAW Hamburg
+ *               2016 INRIA
+ *               2021 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_atmega_common
+ * @ingroup     drivers_periph_gpio_ll
+ * @{
+ *
+ * @file
+ * @brief       Low-level GPIO-NG driver implementation for the ATmega GPIO peripheral
+ *
+ * @author      René Herthel <rene-herthel@outlook.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ * @author      Laurent Navet <laurent.navet@gmail.com>
+ * @author      Robert Hartung <hartung@ibr.cs.tu-bs.de>
+ * @author      Torben Petersen <petersen@ibr.cs.tu-bs.de>
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+
+#include "cpu.h"
+#include "irq.h"
+#include "periph/gpio_ll.h"
+
+static void _set_dir(gpio_port_t port, uint8_t pin, bool output)
+{
+    atmega_gpio_port_t *p = (void *)port;
+    if (output) {
+        p->ddr |= 1U << pin;
+    }
+    else {
+        p-> ddr &= ~(1U << pin);
+    }
+}
+
+static void _set_pull_config(gpio_port_t port, uint8_t pin, gpio_pull_t pull)
+{
+    atmega_gpio_port_t *p = (void *)port;
+    p->port |= pull << pin;
+}
+
+int gpio_ll_init(gpio_port_t port, uint8_t pin, gpio_conf_t *conf)
+{
+    if ((conf->pull > GPIO_PULL_UP)
+        || (conf->state == GPIO_OUTPUT_OPEN_DRAIN))
+        {
```

Comment:
```suggestion
    if ((conf->pull > GPIO_PULL_UP)
        || (conf->state == GPIO_OUTPUT_OPEN_DRAIN)) {
```
or
```suggestion
    if ((conf->pull > GPIO_PULL_UP) ||
        (conf->state == GPIO_OUTPUT_OPEN_DRAIN)) {
```
My personal feeling is that the second one is better readable, but it's probably a question of taste.